src/frontend/src/lib/components/tokens/TokenExchangeBalance.svelte

```diff
@@ -1,21 +1,21 @@
 <script lang="ts">
 	import { isNullish, nonNullish } from '@dfinity/utils';
+	import { i18n } from '$lib/stores/i18n.store';
 	import type { TokenFinancialData } from '$lib/types/token';
 	import { formatUSD } from '$lib/utils/format.utils';
 
 	export let balance: TokenFinancialData['balance'];
 	export let usdBalance: TokenFinancialData['usdBalance'];
+	export let nullishBalanceMessage = '-';
 </script>
 
 <output class="break-all">
 	{#if nonNullish(balance) && nonNullish(usdBalance)}
 		{formatUSD({ value: usdBalance })}
+	{:else if isNullish(balance)}
+		<span class:animate-pulse={isNullish(balance)}>{nullishBalanceMessage}</span>
 	{:else}
-		<span class:animate-pulse={isNullish(balance)}
-			>{formatUSD({
-				value: 0,
-				options: { minFraction: 0, maxFraction: 0 }
-			}).replace('0', '-')}
-		</span>
+		<span class:animate-pulse={isNullish(balance)}>{$i18n.tokens.balance.error.not_applicable}</span
```

Comment:
personally, I would not like to see a bunch of `'n/a'` on the screen... especially in the ICP network, I don't know if it gives a good impression to the users, @StefanBerger-DFINITY

Reply:
@AntonioVentilii at the moment it is defined like this. I guess in case of a change it would be solved with another bug ticket.